packages-new/components/file/src/components/osds-file-item/core/controller.ts

```diff
@@ -0,0 +1,26 @@
+import { OsdsFileItem } from '../osds-file-item';
+
+class OdsFileItemController {
+  constructor(private component: OsdsFileItem) {}
+
+  /**
+   * Attributes validation documentation
+   */
+  validateAttributes(): void {
+    if (!this.component.name) {
+      throw new Error('ods-file-item: name attribute is required');
+    }
+    if (!this.component.size) {
+      throw new Error('ods-file-item: size attribute is required');
+    }
+  }
+  getSizeInKiloBytes(size: number) {
+    const result = Math.round(size / 1000);
```

Comment:
Seems a bit too simple to render human readable size, especially on large files, you could use instead a formatter function like:
```
const byteSizes = ['B', 'KB', 'MB', 'GB', 'TB']; // put this on top outside of controller class

// controller method
function formatBytes(bytes, decimals = 0) {
  const k = 1024;
  const dm = decimals < 0 ? 0 : decimals:
  const i = Math.floor(Math.log(bytes) / Math.log(k));

  return `${parseFloat((bytes / (k ** i)).toFixed(dm))} {byteSizes[i]}`;
}
```